modules/__tests__/routeMatching-test.js

```diff
@@ -139,6 +205,33 @@ describe('A Router', function () {
         done();
       });
     });
+
+    it('matches nested route where parent has no component', function (done) {
+      render(<Router history={new MemoryHistory("/foos/bars")} children={routes} />, div, function () {
+        childRoutes[2].getChildRoutes(null, function (state, barsChildRoutes) {
```

Comment:
Technically, I think this route probably works. It's the same concept as the synchronous version above, which works. I can't figure out how to test the equality of that child route, though.